src/updaters/objectMerge.ts

```diff
@@ -0,0 +1,29 @@
+import { pathName, StatePath } from '../createAction';
+import updater from './updater';
+import { isObject } from '../services/typeService';
+
+/**
+ * Merge an object into another
+ * @example
+ * dispatch(objectMerge(statePath, data))
+ */
+export default (statePath: StatePath, objectToMerge: object) =>
```

Comment:
Can you add a `depthLimit` parameter, that limits the depth of the deep merge? (setting it to 1 would make it shallow). This will give you more control over performance and can be useful when merging data sets, where you'll need to replace the objects in a map at a certain depth. For example with normalized data:
```js
{
  users: {
    1: {},
    2: {},
  },
  comments: {
    /*...*/
  }
}
```
In this example, when merging in an object with the same shape, you'd set the depth limit to 2, if you want to replace user and comment objects that are already there, instead of merging each property.